torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -201,14 +202,18 @@ def apply_tp(
             "attention.wq": colwise_parallel(),
             "attention.wk": colwise_parallel(),
             "attention.wv": colwise_parallel(),
-            "attention.wo": rowwise_parallel(output_layouts=Shard(1)),
+            "attention.wo": rowwise_parallel(
+                output_layouts=Shard(1), use_local_output=False
+            ),
```

Comment:
@XilunWu I think the current CP doesn't handle the case where the attention inputs are DTensor.

Reply:
Yes, I believe CP doesn't handle the case.